scripts/check_names.py

```diff
@@ -1016,15 +1165,21 @@ def main():
     log.addHandler(logging.StreamHandler())
 
     try:
-        code_parser = CodeParser(log)
-        parse_result = code_parser.comprehensive_parse()
+        tf_psa_crypto_code_parser = TFPSACryptoCodeParser(log)
+        parse_result = tf_psa_crypto_code_parser.comprehensive_parse()
+        if build_tree.looks_like_mbedtls_root(os.getcwd()):
+            mbedtls_code_parser = MBEDTLSCodeParser(log)
+            mbedtls_parse_result = mbedtls_code_parser.comprehensive_parse()
+            # Combine parse results together for NameChecker
+            for key in mbedtls_parse_result:
+                parse_result[key] += mbedtls_parse_result[key]
```

Comment:
I would put this logic rather in MBEDTLSCodeParser. MBEDTLSCodeParser parses all mbedtls code, TF-PSA-Crypto code included in 4.x case. Internally it uses TFPSACryptoCodeParser to parse the TF-PSA-Crypto code. And here we have just something like (not tested):
```
if build_tree.looks_like_mbedtls_root(os.getcwd()):
    code_parser = MBEDTLSCodeParser(log)
    parse_result = mbedtls_code_parser.comprehensive_parse()
elsif build_tree.looks_like_tf_psa_crypto_root(os.getcwd()):
    code_parser = TFPSACryptoCodeParser(log)
    parse_result = tf_psa_crypto_code_parser.comprehensive_parse()
```